### PR TITLE
feat(ops): relational ops

### PR DESCRIPTION
**Pull request by littledivy**
Join two independent ops into one. A fast impl of one + a slow callback of another. Here's an example showing optimized paths for latin-1 via fast call and the next-best fallback using V8 apis.

```rust
#[op(v8)]
fn op_encoding_encode_into_fallback(
  scope: &mut v8::HandleScope,
  input: serde_v8::Value,
// ...

#[op(fast, slow = op_encoding_encode_into_fallback)]
fn op_encoding_encode_into(
  input: Cow<'_, str>,
// ...
```

Benchmark results of the fallback path:
```
time target/release/deno run -A --unstable ./cli/tests/testdata/benches/text_encoder_into_perf.js
________________________________________________________
Executed in   70.90 millis    fish           external
   usr time   57.76 millis    0.23 millis   57.53 millis
   sys time   17.02 millis    1.28 millis   15.74 millis

target/release/deno_main run -A --unstable ./cli/tests/testdata/benches/text_encoder_into_perf.js
________________________________________________________
Executed in  154.00 millis    fish           external
   usr time   67.14 millis    0.26 millis   66.88 millis
   sys time   38.82 millis    1.47 millis   37.35 millis
```

### CODE REVIEW

File: ops/attrs.rs [hunk 1]
```diff
@@ -1,44 +1,61 @@
...
+          let _ = input.parse::<Token![,]>();
+        }
+        Err(_) => {
+          break;
+        }
       }
     }
 
-    let is_wasm = vars.contains(&"wasm");
+    self_.must_be_fast = self_.is_wasm || fast;
```
Comment (aapoalas) [inline ops/attrs.rs:57]: I guess this is still not being asserted anywhere?

File: ops/lib.rs [hunk 2]
```diff
@@ -111,24 +111,68 @@ impl Op {
...
+          pub fn decl #generics () -> #core::OpDecl #where_clause {
+            #core::OpDecl {
+              name: Self::name(),
+              v8_fn_ptr: Self::v8_fn_ptr::<#type_params>(),
+              enabled: true,
+              fast_fn: #decl,
+              is_async: #is_async,
+              is_unstable: #is_unstable,
+              is_v8: #is_v8,
+              argc: 0,
```
Comment (aapoalas) [inline ops/lib.rs:148]: argc 0 seems suspect?

File: ops/lib.rs [hunk 3]
```diff
@@ -111,24 +111,68 @@ impl Op {
...
+        }
+
+        #impl_and_fn
+      };
+    }
+
     let has_fallible_fast_call = active && optimizer.returns_result;
 
     let (v8_body, argc) = if is_async {
+      let deferred = attrs.deferred;
```
Comment (aapoalas) [inline ops/lib.rs:164]: Is this extraction necessary?

Comment (littledivy) [reply ops/lib.rs:164]: yes, `Attributes` doesn't impl Copy anymore

File: ops/lib.rs (hunk 2 above)
Comment (littledivy) [reply ops/lib.rs:148]: i don't think we need argc anymore since we reverted async op codegen. I'll open a PR to remove it.

**Review by aapoalas — APPROVED** [review]
LGTM

File: ops/attrs.rs (hunk 1 above)
Comment (littledivy) [reply ops/attrs.rs:50]: hmm, it is handled in the optimizer: https://github.com/denoland/deno/blob/888ceac7fdc6e9b96d860f183dd2f31df3f3d601/ops/optimizer.rs#L368-L369

doesn't look like `fast_compatible` is asserted though anywhere.

**Pull request merged**